### ISSUE TITLE
Validate v8 build with .NET 8

### ISSUE THREAD

**Issue by martincostello** [App-vNext/Polly#1141]
Add a build configuration to validate in CI that Polly v8 works as expected with .NET 8 with a `net8.0` TFM using the prerelease packages.

Once preview 4 is available, remove the internal `TimeProvider` copy and use the version shipping in a NuGet package instead.

Once .NET 8 ships as stable in November, we can then roll `net8.0` into the main build setup and ship a package that explicitly supports it.

**Comment by martintmk** [comment]
As a part of this task we should also unify all `Microsoft.Extensions.*` and `System.*` packages to version 8.0.0. These versions should fix known compatibility problems seen in versions. Would simplify our setup too as we don't have to deal with different versions for each framework anymore.

**Comment by martincostello** [comment]
Yep I'll sort that out once it's available (that's p4 too right?). For now I've just extended the multi-targeting while I'm playing with it locally.

**Comment by martintmk** [comment]
> that's p4 too right?

Hopefully yes, we just need to wait until the `System.Bcl.TimeProvider` package appears in nuget :)

**Comment by martintmk** [comment]
@martincostello I came across this post:

https://devblogs.microsoft.com/dotnet/package-validation/

I think it's something we can do for V8 too! ;)

**Comment by martincostello** [comment]
Yep it would be good to add that too - I already use it in a number of my OSS projects. We should be able to enable it separate from the .NET 8 upgrade.

**Comment by martincostello** [comment]
Added package validation in #1172. There's also some additional checking already as part of #1134.

**Comment by martincostello** [comment]
Completed by #1144.
